### PR TITLE
[corlib] Move the implementations of virtual methods from Assembly to RuntimeAssembly to ease netcore integration.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mcs/class/corlib/System.Reflection/Assembly.cs [hunk 1]
```diff
@@ -207,33 +177,20 @@ internal Evidence UnprotectedGetEvidence ()
...
-				return loc;
+				throw new NotImplementedException ();
 			}
 		}
 
 		[ComVisible (false)]
 		public virtual string ImageRuntimeVersion {
 			get {
-				return InternalImageRuntimeVersion ();
+				throw new NotImplementedException ();
```
Comment (lambdageek) [inline mcs/class/corlib/System.Reflection/Assembly.cs]: `AssemblyBuilder.ImageRuntimeVersion` calls this one.

File: mcs/class/corlib/System.Reflection/Assembly.cs [hunk 2]
```diff
@@ -244,113 +201,42 @@ public virtual void GetObjectData (SerializationInfo info, StreamingContext cont
 
 		public virtual bool IsDefined (Type attributeType, bool inherit)
 		{
-			return MonoCustomAttrs.IsDefined (this, attributeType, inherit);
+			throw new NotImplementedException ();
```
Comment (lambdageek) [inline mcs/class/corlib/System.Reflection/Assembly.cs]: `AssemblyBuilder.IsDefined` calls this one explicitly.

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
This causes a regression in a sys.web test:
MonoTests.System.Web.Caching.OutputCacheTest.Serialize
The first commit seems to cause it.

**Comment by marek-safar** [comment]
It's ok to regenerate the expected byte stream. We should have all the fields as marked as NonSerialized in Assembly class long time ago

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**